pkg/controller/secrets/cluster_ca_controller.go

```diff
@@ -0,0 +1,111 @@
+// limitations under the License.
+
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/certificatemanager"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/utils"
+	rcertificatemanagement "github.com/tigera/operator/pkg/render/certificatemanagement"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+type ClusterCAController struct {
+	client        client.Client
+	scheme        *runtime.Scheme
+	clusterDomain string
+	multiTenant   bool
+	log           logr.Logger
+}
+
+func AddClusterCAController(mgr manager.Manager, opts options.AddOptions) error {
+	r := &ClusterCAController{
+		client:        mgr.GetClient(),
+		scheme:        mgr.GetScheme(),
+		clusterDomain: opts.ClusterDomain,
+		log:           logf.Log.WithName("controller_cluster_ca"),
+	}
+
+	// Create a controller using the reconciler and register it with the manager to receive reconcile calls.
+	c, err := controller.New("cluster-ca-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for triggers.
+	if err = c.Watch(&source.Kind{Type: &operatorv1.Installation{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("cluster-ca-controller failed to watch primary resource: %w", err)
+	}
+	if err = utils.AddSecretsWatch(c, certificatemanagement.CASecretName, common.OperatorNamespace()); err != nil {
+		return fmt.Errorf("cluster-ca-controller failed to watch CA secret: %w", err)
+	}
+
+	return nil
+}
+
+func (r *ClusterCAController) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	logc := r.log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	// Get Installation resource.
+	instance := &operatorv1.Installation{}
+	if err := r.client.Get(ctx, utils.DefaultInstanceKey, instance); err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Create the cluster CA. This is done implicitly by initializing a certificate manager instance
+	// and passing the "AllowCACreation" option.
+	opts := []certificatemanager.Option{
+		certificatemanager.AllowCACreation(),
+		certificatemanager.WithLogger(logc),
+	}
+	cm, err := certificatemanager.Create(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace(), opts...)
```

Comment:
Could you add a description on what this CA is used for, just to help with understanding. I'm sort of guessing, but it's used for non tenant specific communication in the management cluster, is that right? Any component that's tenanted would use the CA created by the tenant controller, is that correct?